alacritty/src/daemon.rs

```diff
@@ -43,18 +49,43 @@ where
         .map(|_| ())
 }
 
+/// Get working directory of controlling process.
+#[cfg(not(windows))]
+pub fn foreground_process_path() -> Result<PathBuf, Box<dyn Error>> {
+    let mut pid = unsafe { libc::tcgetpgrp(tty::master_fd()) };
+    if pid < 0 {
+        pid = tty::child_pid();
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    let link_path = format!("/proc/{}/cwd", pid);
+    #[cfg(target_os = "freebsd")]
+    let link_path = format!("/compat/linux/proc/{}/cwd", pid);
+
+    #[cfg(not(target_os = "macos"))]
+    let cwd = fs::read_link(link_path)?;
+
+    #[cfg(target_os = "macos")]
+    let cwd = macos::proc::cwd(pid)?;
+
+    Ok(cwd)
+}
+
 #[cfg(not(windows))]
 fn spawn_daemon<I, S>(program: &str, args: I) -> io::Result<()>
 where
     I: IntoIterator<Item = S> + Copy,
     S: AsRef<OsStr>,
 {
+    let mut command = Command::new(program);
+    let mut command_builder = command.args(args);
+    if let Ok(cwd) = foreground_process_path() {
+        command_builder.current_dir(cwd);
+    }
+    command_builder =
+        command_builder.stdin(Stdio::null()).stdout(Stdio::null()).stderr(Stdio::null());
```

Comment:
I think we can just rename this variable to `builder` without any problems. That way we get rid of the ugly linebreak.